pkg/encoding/fix.go

```diff
@@ -47,3 +53,18 @@ func fixPersistentVolumes(app *spec.App) error {
 	}
 	return nil
 }
+
+func fixConfigData(app *spec.App) error {
+	// if only one configData is defined and it's name is not specified
+	if len(app.ConfigData) == 1 && app.ConfigData[0].Name == "" {
```

Comment:
I think this logic is more straightforward than the one, where we are checking if the *length is 1 && no name* given then we default.
Otherwise we iterate on all of items to check if name is given!